app/domain/authentication/update_authenticator_config.rb

```diff
@@ -0,0 +1,60 @@
+require 'command_class'
+
+module Authentication
+
+  UpdateAuthenticatorConfig = CommandClass.new(
+    dependencies: {
+      webservice_class: ::Authentication::Webservice,
+      resource_class: ::Resource,
+      authenticator_config_class: ::AuthenticatorConfig
+    },
+    inputs: %i(account authenticator service_id enabled current_user)
+  ) do
+    
+    def call
+      validate_resource_visible
+      validate_resource_is_authenticator
+      validate_resource_writable
+
+      update_authenticator_config
+    end
+
+    private
+
+    def validate_resource_visible
+      raise Exceptions::RecordNotFound, resource_id \
+        unless resource.visible_to?(@current_user)
+    end
+
+    def validate_resource_is_authenticator
```

Comment:
can we change this method name to `validate_webservice_exists`? it seems that this is what it does

Reply:
And if this is really what's happening, consider using [ValidateWebserviceExists](https://github.com/cyberark/conjur/blob/master/app/domain/authentication/validate_webservice_exists.rb)

You can see a usage [here](https://github.com/cyberark/conjur/blob/master/app/domain/authentication/validate_status.rb)

Reply:
The purpose of this method is to validate that the webservice is a configured authenticator instead of (for example) a nested status webservice. Without this check you can configure the nested status webservice, which doesn't really make sense. I'll change the name to be more explicit.